src/hammer-tech/stackup.py

```diff
@@ -0,0 +1,230 @@
+                # we are asking for a pitch that is width-constrained
+                width = self.snap(second.width_at_least - (self.grid_unit()*2))
+                spacing = self.snap(pitch - width)
+        return spacing
+
+    # This method will return the maximum width a wire can be to consume a given number of routing tracks.
+    # This assumes the neighbors of the wide wire are minimum-width routes.
+    # i.e. T W T
+    # T = thin / min-width
+    # W = wide
+    # Returns width, spacing and start
+    # See min_spacing_from_pitch documentation for an explanation of the calculation.
+    def get_width_spacing_start_twt(self, tracks: int) -> Tuple[float, float, float]:
+        ws = self.power_strap_widths_and_spacings
+        spacing = ws[0].min_spacing
+        # the T W T pattern contains one wires (W) and 2 spaces (S2)
+        s2w = self.snap((tracks + 1) * self.pitch - self.min_width)
+        assert (int(self.snap(s2w / self.grid_unit())) % 2 == 0), "This calculation should always produce an even s2w"
+        width = self.snap(s2w - spacing*2)
+        for first, second in zip(ws[:-1], ws[1:]):
+            if s2w >= (second.min_spacing*2 + second.width_at_least):
+                spacing = second.min_spacing
+                width = self.snap(s2w - spacing*2)
+            elif s2w >= (first.min_spacing*2 + second.width_at_least):
+                # we are asking for a pitch that is width-constrained
+                if (int(second.width_at_least / self.grid_unit()) % 2 == 0):
+                    # even
+                    width = self.snap(second.width_at_least - (self.grid_unit()*2))
+                else:
+                    # odd
+                    width = self.snap(second.width_at_least - self.grid_unit())
+                spacing = self.snap((s2w - width)/2.0)
+        assert (int(self.snap(self.min_width / self.grid_unit())) % 2 == 0), "Assuming all min widths are even here, if not fix me"
+        assert (int(self.snap(width / self.grid_unit())) % 2 == 0), "This calculation should always produce an even width"
+        start = self.snap(self.min_width/2.0 + spacing)
+        return (width, spacing, start)
+
+    # This method will return the maximum width a wire can be to consume a given number of routing tracks
+    # This assumes one neighbor is a min width wire, and the other is the same size as this (mirrored)
+    # i.e. T W W T
+    # T = thin / min-width
+    # W = wide
+    # Returns width, spacing, and start
+    # See min_spacing_from_pitch documentation for an explanation of the calculation
+    # force_even: True to force the strap width to be an even multiple of the grid unit
+    def get_width_spacing_start_twwt(self, tracks: int, force_even: bool = False) -> Tuple[float, float, float]:
+        ws = self.power_strap_widths_and_spacings
+        spacing = ws[0].min_spacing
+        # the T W W T pattern contains two wires (W2) and 3 spaces (S3)
```

Comment:
I think this needs to be slightly more clear given the next line.
Does s3w2 include all the wires and spaces above?
It doesn't appear to but I'm not exactly sure what is excluded.
I initially thought it would hold this: 
`1minW + 1Space + tPitches + 1Space + tPitches + 1Space + 1minW = (2t+2)Pitches+1Space`
assuming `Pitch = 1Space + 1 minW`
but instead it seems to hold `(2t+1)Pitches - 1minW` so you would be exlcluding the first and last `1minW +1Space` from above.
So s3w2 holds just the distance from the left hand side of the first Wide wire to the right hand side of the second Wide wire?
Do I understand this correctly?

Reply:
Still missing a space. I'll draw it for you.

Reply:
Resolved offline